### ISSUE TITLE
`--silent` also hide diagnostics

### ISSUE THREAD

**Issue by nikaro** [rvben/rumdl#141]
```sh
⏺ /v/f/2/_/T/1owYTu > cat test-bad.md
# HEAD 1
body
⏺ /v/f/2/_/T/1owYTu > rumdl check test-bad.md
test-bad.md:1:1: [MD022] Expected 1 blank line below heading [*]

Issues: Found 1 issues in 1 file (0ms)
Run `rumdl fmt` to automatically fix 1 of the 1 issues
⏺ /v/f/2/_/T/1owYTu > rumdl check --silent test-bad.md
⏺ /v/f/2/_/T/1owYTu >
```

**Comment by nikaro** [comment]
Isn't the issue that it implies `--quiet`?

- https://github.com/rvben/rumdl/blob/main/src/main.rs#L444
- https://github.com/rvben/rumdl/blob/main/src/main.rs#L1362

**Comment by rvben** [comment]
Hi @nikaro! Thank you for reporting this. The --silent flag was hiding diagnostics when it should only suppress the summary/progress output.

I've fixed this to match Ruff's behavior:

- --quiet: Print diagnostics, but nothing else (no summary, no progress messages)
- --silent: Disable all logging (but still exit with status code upon detecting diagnostics)

# Demonstration

```bash
$ cat test-bad.md
# HEAD 1
body
```

## Normal mode - shows diagnostics AND summary

```bash
$ rumdl check test-bad.md
test-bad.md:1:1: [MD022] Expected 1 blank line below heading [*]

Issues: Found 1 issues in 1 file (0ms)
Run `rumdl fmt` to automatically fix 1 of the 1 issues
```

## --quiet mode - shows ONLY diagnostics (no summary)

```bash
$ rumdl check --quiet test-bad.md
test-bad.md:1:1: [MD022] Expected 1 blank line below heading [*]
```

## --silent mode - shows NOTHING (but still exits with code 1)

```bash
$ rumdl check --silent test-bad.md
$ echo $?
1
```

So now if you want to see diagnostics without the summary footer, you can use --quiet. And if you want absolutely no output (like for CI scripts that only care about the exit code), you can use --silent.

This will land in the next release. Thanks again! 🚀
